src/sql/workbench/parts/dataExplorer/common/nodeContextUtils.ts

```diff
@@ -0,0 +1,104 @@
+import { Disposable } from 'vs/base/common/lifecycle';
+import { IConnectionManagementService } from 'sql/platform/connection/common/connectionManagement';
+import { ConnectionProfile } from 'sql/platform/connection/common/connectionProfile';
+import { ICapabilitiesService } from 'sql/platform/capabilities/common/capabilitiesService';
+import { mssqlProviderName } from 'sql/platform/connection/common/constants';
+import { NodeType } from 'sql/workbench/parts/objectExplorer/common/nodeType';
+
+export class NodeContextUtils extends Disposable {
+
+	static IsMssqlProvided = new RawContextKey<boolean>('isMssqlProvided', false);
+	static IsDatabaseOrServer = new RawContextKey<boolean>('isDatabaseOrServer', false);
+	static IsWindows = new RawContextKey<boolean>('isWindows', os.platform() === 'win32');
+	static IsCloud = new RawContextKey<boolean>('isCloud', false);
+	static NodeType = new RawContextKey<string>('nodeType', undefined);
+	static NodeLabel = new RawContextKey<string>('nodeLabel', undefined);
+
+	private isMssqlProvidedKey: IContextKey<boolean>;
+	private isCloudKey: IContextKey<boolean>;
+	private nodeTypeKey: IContextKey<string>;
+	private nodeLabelKey: IContextKey<string>;
+	private isDatabaseOrServerKey: IContextKey<boolean>;
+
+	constructor(
+		private nodeContextValue: INodeContextValue,
+		@IContextKeyService private contextKeyService: IContextKeyService,
+		@IConnectionManagementService private connectionManagementService: IConnectionManagementService,
+		@ICapabilitiesService private capabilitiesService: ICapabilitiesService
+	) {
+		super();
+		this.bindContextKeys();
+
+		// Set additional node context keys
+		if (this.nodeContextValue.node) {
+			const node = this.nodeContextValue.node;
+			if (node.payload) {
+				this.isProvidedByMssql();
+				this.isCloud();
+				if (node.contextValue && node.providerHandle === mssqlProviderName) {
+					this.isDatabaseOrServer();
+					this.nodeTypeKey.set(node.contextValue);
+				} else if (node.type) {
+					this.isDatabaseOrServer();
+					this.nodeTypeKey.set(node.type);
+				}
+			}
+			if (node.label) {
+				this.nodeLabelKey.set(node.label.label);
+			}
+		}
+	}
+
+	private bindContextKeys(): void {
+		this.isMssqlProvidedKey = NodeContextUtils.IsMssqlProvided.bindTo(this.contextKeyService);
+		this.isCloudKey = NodeContextUtils.IsCloud.bindTo(this.contextKeyService);
+		this.nodeTypeKey = NodeContextUtils.NodeType.bindTo(this.contextKeyService);
+		this.nodeLabelKey = NodeContextUtils.NodeLabel.bindTo(this.contextKeyService);
+		this.isDatabaseOrServerKey = NodeContextUtils.IsDatabaseOrServer.bindTo(this.contextKeyService);
+	}
+
+	/**
+	 * Helper function to tell whether node provider is MSSQL
+	 */
+	private isProvidedByMssql(): void {
+		if (this.nodeContextValue.node.payload.providerName === 'MSSQL') {
+			this.isMssqlProvidedKey.set(true);
+		}
+	}
+
+	/**
+	 * Helper function to tell whether a connected node is cloud or not
+	 */
+	private isCloud(): void {
```

Comment:
>isCloud [](start = 9, length = 7)

Prefer we don't name functions is<Noun> and then not return anything. something like setIsCloud would be better IMO (same with other methods in this file).